public/resources/functions/inqueue-callback.protected.js

```diff
@@ -61,22 +61,24 @@ exports.handler = function (context, event, callback) {
   //   }
   // }
 
-  //  find the task given the callSid - get TaskSid
-  async function getTask(callSid) {
-    attrFilter = `call_sid=  '${callSid}'`;
-
+  //  find the task given the callSid or the task sid - get TaskSid
+  async function getTask(sid) {
     try {
-      let task = await client.taskrouter
-        .workspaces(context.TWILIO_WORKSPACE_SID)
-        .tasks.list({
-          evaluateTaskAttributes: attrFilter,
-          limit: 20,
-        });
+      let result = await (sid.startsWith('CA') 
```

Comment:
Thanks @ktalebian ! In the future the idea is to have `getTask()` as an helper since it's used by three different functions. 

The implementation I'd lean towards is the one here:  https://github.com/twilio-labs/plugin-queued-callbacks-and-voicemail/blob/071d8cc5a36f26faad8cac128277ff4b79fbaf59/public/resources/functions/queue-menu.protected.js#L118

The reason why I didn't do that yet, it's because I want to minimise the number of changes. Let me know what you think of that implementation.

Reply:
Solved in 228cd02